cognite/power/data_classes.py

```diff
@@ -610,25 +586,28 @@ def relationships(
             raise ValueError("Can not combine _sources and _targets.")
         if not _source_external_ids and not _target_external_ids:
             return PowerAssetList([], cognite_client=self._cognite_client)
-        rels = self._cognite_client.relationships.list(
-            source_external_ids=_source_external_ids,
-            target_external_ids=_target_external_ids,
-            labels=LabelFilter(contains_all=[label]),
-            limit=None,
+        assets = self._cognite_client.power_assets.list(
+            grid_type=grid_type, base_voltage=base_voltage, **{"metadata": {"type": power_type}}
         )
         if _source_external_ids:
-            asset_ids = [r.target_external_id for r in rels]
+            _target_external_ids = [a.external_id for a in assets]
+            rels = self._cognite_client.relationships.list(
+                source_external_ids=_source_external_ids,
+                target_external_ids=_target_external_ids,
+                labels=LabelFilter(contains_all=[label]),
+                limit=None,
+            )
+            assets = [a for a in assets if a.external_id in [r.target_external_id for r in rels]]
```

Comment:
`target_external_ids = {r.target_external_id for r in rels}`
and then in (set) here please